fix: use correct release type in upgrade process

jacderida opened the pull request:
The faucet and daemon upgrades did not work correctly because the release type was hard coded to `ReleaseType::Safenode` and was not changed when this code was refactored.

## Description

reviewpad:summary

bochaco commented on sn_interface/src/messaging/data/mod.rs:
```diff
@@ -247,6 +247,24 @@ impl QueryResponse {
         }
     }
 
+    /// Returns true if the result returned is DataNotFound
```
I think this is cleaner?
```
matches!(
            self,
            GetChunk(Err(ErrorMsg::DataNotFound(_)))
                | GetRegister((Err(ErrorMsg::DataNotFound(_)), _))
                | GetRegisterEntry((Err(ErrorMsg::DataNotFound(_)), _))
                | GetRegisterOwner((Err(ErrorMsg::DataNotFound(_)), _))
                | ReadRegister((Err(ErrorMsg::DataNotFound(_)), _))
                | GetRegisterPolicy((Err(ErrorMsg::DataNotFound(_)), _))
                | GetRegisterUserPermissions((Err(ErrorMsg::DataNotFound(_)), _))
                | SpentProofShares((Err(ErrorMsg::DataNotFound(_)), _))
        )
```

bochaco replied:
@joshuef my suggestion was for the `is_data_not_found` function rather than the `is_success` which was ok but now will return the opposite value

bochaco commented on sn_client/src/api/queries.rs:
```diff
@@ -93,7 +95,23 @@ impl Client {
-                Ok(Ok(query_result)) => break Ok(query_result),
+                Ok(Ok(query_result)) => {
+                    if !query_result.is_data_not_found() {
+                        data_not_found_count += 1;
```
The var name seems to imply the condition is incorrectly negated?

joshuef replied:
ups, yeh i shuffled the order but forgot to remove the `!`. good one!

bochaco commented on sn_client/src/api/queries.rs:
```diff
@@ -93,7 +95,23 @@ impl Client {
+                    if !query_result.is_data_not_found() {
+                        data_not_found_count += 1;
+                        // if we've tested enough adults, we can return failure
+                        // we do this twice to be sure connection issues were not involved
```
Do we mean here connection issues between Elders and Adults? I assume connection issues from client to Elders won't return a data-not-found error, correct?

joshuef replied:
yeh, issues communicating to elders / a timeout/cleanup issue eg. I'll reword here

joshuef replied:
Actually this is wrong (I think i said this yesterday in the meet, and then forgot about it 🤦 ). 

We're _only_ tracking `DataNotFound` here. Connection errors dont come into it. I've updated it to track each index's DataNotFound... and we'd loop only until we have DataNotFound at each of them

bochaco commented:
bors r+

bochaco commented:
bors r+

jacderida committed:
fix: use correct release type in upgrade process

The faucet and daemon upgrades did not work correctly because the release type was hard coded to
`ReleaseType::Safenode` and was not changed when this code was refactored.
